### PR TITLE
Make the fields of RequestInit optional.

### PR DESCRIPTION
**Pull request by zakpatterson**
This way, users do not have to recite every field when creating a `RequestInit`. Also, the spec says the `RequestInit` dictionary must contain a field `window` which is set to null. We might as well include this too as a final variable, since the spec shows omitting it should be a type error.

Prompted by gitter conversation here: https://gitter.im/scala-js/scala-js?at=5d80415105fd3716953afa47

### CODE REVIEW

**Comment by zakpatterson** [comment]
I see, however that `@js.native` traits are not supposed to have default definitions: https://www.scala-js.org/doc/interoperability/facade-types.html.  Oddly this works with scala.js 1.0 and 0.6 on scala 2.12 and 2.13, but fails on 2.11 and 2[.yikes].10.

**Comment by flavomulti** [comment]
I think if you pull off the js.native this will work as expected though. Are you going to resubmit?

**Comment by zakpatterson** [comment]
@flavomulti I just removed the js.native as you suggested. I remember now that I added that so that the window field could be set to null explicitly. Perhaps the way I have it now will work?

**Comment by flavomulti** [comment]
I think that as long as its not set, e.g. js.undefined, its ok not to specify it. If you do specify window, it must be null...something like that. It's never clear why funky things like this exist, but it is what it is.

**Comment by zakpatterson** [comment]
Ah, that makes sense. So this will still throw a typeerror if a user tries to set it to something that's not null, which is what the spec wants. Ok thanks @flavomulti !

**Comment by aappddeevv** [comment]
Is there anyway to get this merged? With 1.0 out, we really need js.undefined on the RequestInit trait's properties as it is very unergonomic to use otherwise.

File: src/main/scala/org/scalajs/dom/experimental/Fetch.scala [hunk 1]
```diff
@@ -136,27 +136,27 @@ object RequestInit {
 trait RequestInit extends js.Object {
   var method: js.UndefOr[HttpMethod]
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/experimental/Fetch.scala:137]: `method` should also have `= js.undefined`. It seems optional, according to step 27 of the constructor of `Request` ([spec](https://fetch.spec.whatwg.org/#request-class)).

Comment (zakpatterson) [reply src/main/scala/org/scalajs/dom/experimental/Fetch.scala]: 55c1d66

**Comment by sjrd** [comment]
Thank you @zakpatterson

**Comment by aappddeevv** [comment]
Thanks!

**Pull request merged**